tests/ops/qutrit/test_qutrit_parametric_ops.py

```diff
@@ -0,0 +1,477 @@
+    qml.ControlledQutritUnitary(TCLOCK, wires=[0], control_wires=[2]),
+]
+
+BROADCASTED_OPERATIONS = [
+    qml.TRX(np.array([0.142, -0.61, 2.3]), wires=0, subspace=[1, 2]),
+    qml.QutritUnitary(np.array([TSHIFT, TCLOCK]), wires=0),
+    qml.ControlledQutritUnitary(np.array([TSHIFT, TCLOCK]), wires=[0], control_wires=[2]),
+]
+
+NON_PARAMETRIZED_OPERATIONS = [
+    qml.TShift(wires=0),
+    qml.TClock(wires=0),
+    qml.TAdd(wires=[0, 1]),
+    qml.TSWAP(wires=[0, 1]),
+]
+
+
+ALL_OPERATIONS = NON_PARAMETRIZED_OPERATIONS + PARAMETRIZED_OPERATIONS
+
+dot_broadcasted = lambda a, b: np.einsum("...ij,...jk->...ik", a, b)
+multi_dot_broadcasted = lambda matrices: reduce(dot_broadcasted, matrices)
+
+
+class TestOperations:
+    @pytest.mark.parametrize("op", ALL_OPERATIONS + BROADCASTED_OPERATIONS)
+    def test_parametrized_op_copy(self, op, tol):
+        """Tests that copied parametrized ops function as expected"""
+        copied_op = copy.copy(op)
+        assert np.allclose(op.matrix(), copied_op.matrix(), atol=tol)
+
+        op = qml.adjoint(op)
+        copied_op2 = copy.copy(op)
+        assert np.allclose(op.matrix(), copied_op2.matrix(), atol=tol)
+
+    @pytest.mark.parametrize("op", PARAMETRIZED_OPERATIONS)
+    def test_adjoint_unitaries(self, op, tol):
+        """Test that matrices of adjoint operations behave correctly"""
+        op_d = op.adjoint()
+        res1 = np.dot(op.matrix(), op_d.matrix())
+        res2 = np.dot(op_d.matrix(), op.matrix())
+        assert np.allclose(res1, np.eye(3 ** len(op.wires)), atol=tol)
+        assert np.allclose(res2, np.eye(3 ** len(op.wires)), atol=tol)
+        assert op.wires == op_d.wires
+
+    @pytest.mark.parametrize("op", BROADCASTED_OPERATIONS)
+    def test_adjoint_unitaries_broadcasted(self, op, tol):
+        """Test that matrices of adjoint operations with broadcasting behave correctly"""
+        op_d = op.adjoint()
+        res1 = dot_broadcasted(op.matrix(), op_d.matrix())
+        res2 = dot_broadcasted(op_d.matrix(), op.matrix())
+        I = [np.eye(3 ** len(op.wires))] * op.batch_size
+        assert np.allclose(res1, I, atol=tol)
+        assert np.allclose(res2, I, atol=tol)
+        assert op.wires == op_d.wires
+
+
+class TestParameterFrequencies:
+    @pytest.mark.parametrize("op", PARAMETRIZED_OPERATIONS)
+    def test_parameter_frequencies_match_generator(self, op, tol):
+        """Check that parameter frequencies of parametrized operations are defined correctly."""
+        if not qml.operation.has_gen(op):
+            pytest.skip(f"Operation {op.name} does not have a generator defined to test against.")
+
+        gen = op.generator()
+        mat = gen.matrix()
+
+        gen_eigvals = np.round(np.linalg.eigvalsh(mat), 8)
+        freqs_from_gen = qml.gradients.eigvals_to_frequencies(tuple(gen_eigvals))
+
+        freqs = op.parameter_frequencies
+        assert np.allclose(freqs, freqs_from_gen, atol=tol)
+
+
+# TODO: Add tests for decompositions
```

Comment:
todo in this PR?

Reply:
Nope. I'm not even sure if decompositions are going to happen considering there isn't any hardware to use any of these gates :)